pkg/catalog/uncommitted_iterator.go

```diff
@@ -0,0 +1,127 @@
+package catalog
+
+import (
+	"context"
+
+	"github.com/treeverse/lakefs/pkg/graveler"
+)
+
+type UncommittedIterator struct {
+	store     Store
+	ctx       context.Context
+	err       error
+	branchItr graveler.BranchIterator
+	entryItr  *valueEntryIterator
+	entry     *UncommittedRecord
+}
+
+type UncommittedRecord struct {
+	branchID graveler.BranchID
+	*EntryRecord
+}
+
+func NewUncommittedIterator(ctx context.Context, store Store, repository *graveler.RepositoryRecord) (*UncommittedIterator, error) {
+	bItr, err := store.ListBranches(ctx, repository)
+	if err != nil {
+		return nil, err
+	}
+	return &UncommittedIterator{
+		store:     store,
+		ctx:       ctx,
+		branchItr: bItr,
+	}, nil
+}
+
+func (u *UncommittedIterator) nextStaging() bool {
+	if u.entryItr != nil {
+		u.entryItr.Close()
+	}
+	vItr, err := u.store.ListStaging(u.ctx, u.branchItr.Value().Branch)
+	if err != nil {
+		u.err = err
+		return false
+	}
+	u.entryItr = NewValueToEntryIterator(vItr)
+	return true
+}
+
+func (u *UncommittedIterator) next() bool {
```

Comment:
It was hard for me to understand why this function is correct. Consider renaming and/or document it.